specutils/utils/spectral_region.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import astropy.units as u
+
+
+class SpectralRegion:
+    """
+    SpectralRegion is currently a container class enables some simplicty
+    in defining and passing a region (interval) for a spectrum.
+
+    In the future, there might be more functionality added in here and there
+    is some discussion that this might/could move to astropy.region.
+    """
+
+    def __init__(self, lower, upper):
+        """
+        Lower and upper values for the interval.
+
+        Parameter
+        --------
+
+        lower: number with ``astropy.units`` unit
+           The lower bound of the region.
+
+        upper: number with ``astropy.units`` unit
+           The upper bound of the region.
+
+        Notes
+        -----
+
+        The ``astropy.units`` for each do not have to be the same, but to retain
+        sanity it might be easiest to have them the same.  The will need to be
```

Comment:
I don't think you want to say the "retain sanity" part - there's no reason *not* to do this, and for some science cases it may be useful. (e.g., a cosmic ray where you know the location in pixels, but the other side you want your standard wavelength window for the line you're measuring).

Also I'd say the "dispersion as opposed to flux" should go in the description of the parameter as I noted above.

Reply:
I removed the note altogether as it really isn't needed.